test/units/modules/network/vyos/test_vyos_firewall_interfaces.py

```diff
@@ -0,0 +1,296 @@
+        ]
+        self.execute_module(changed=True, commands=commands)
+
+    def test_vyos_firewall_rule_set_04_deleted_idem(self):
+        set_module_args(
+            dict(config=[dict(name='eth1'),
+                         dict(name='eth3')],
+                 state="deleted"))
+        self.execute_module(changed=False, commands=[])
+
+    def test_vyos_firewall_rule_set_02_replaced_idem(self):
+        set_module_args(
+            dict(config=[
+                dict(name='eth0',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[
+                                  dict(name='INBOUND', direction='in'),
+                                  dict(name='OUTBOUND', direction='out'),
+                                  dict(name='LOCAL', direction='local')
+                              ]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ]),
+                dict(name='eth2',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[
+                                  dict(name='INBOUND', direction='in'),
+                                  dict(name='OUTBOUND', direction='out'),
+                                  dict(name='LOCAL', direction='local')
+                              ]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ])
+            ],
+                state="replaced"))
+        self.execute_module(changed=False, commands=[])
+
+    def test_vyos_firewall_rule_set_01_replaced(self):
+        set_module_args(
+            dict(config=[
+                dict(name='eth0',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[
+                                  dict(name='INBOUND', direction='in'),
+                              ]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ]),
+                dict(name='eth2',
+                     access_rules=[
+                         dict(afi='ipv4',
+                              rules=[dict(name='LOCAL', direction='local')]),
+                         dict(afi='ipv6',
+                              rules=[dict(name='V6-LOCAL', direction='local')])
+                     ])
+            ],
+                state="replaced"))
+        commands = [
+            "delete interfaces ethernet eth0 firewall out name",
+            "delete interfaces ethernet eth0 firewall local name",
+            "delete interfaces ethernet eth2 firewall in name",
+            "delete interfaces ethernet eth2 firewall out name",
+        ]
+        self.execute_module(changed=True, commands=commands)
```

Comment:
This testcase for 'replaced' state , consists of only deletion commands. Can another testcase be added, which includes addition of rules?

Reply:
Done